File: crates/laddu-amplitudes/src/angular.rs
//! Angular, barrier, and density-matrix factor amplitudes.

mod barrier;
mod harmonics;
mod sdme;
#[cfg(test)]
mod tests;
mod wigner;

pub use barrier::BlattWeisskopf;
pub use harmonics::{PolPhase, Ylm, Zlm};
pub use sdme::{PhotonHelicity, PhotonPolarization, PhotonSDME};
pub use wigner::WignerD;

File: py-laddu/laddu/amplitudes/angular.py
"""Angular amplitudes and related factors."""

from fractions import Fraction
from typing import TypeAlias

from laddu.laddu import (
    BlattWeisskopf,
    ClebschGordan,
    PhotonSDME,
    PolPhase,
    Wigner3j,
    WignerD,
    Ylm,
    Zlm,
)

QuantumNumber: TypeAlias = int | float | Fraction

__all__ = [
    'BlattWeisskopf',
    'ClebschGordan',
    'PhotonSDME',
    'PolPhase',
    'QuantumNumber',
    'Wigner3j',
    'WignerD',
    'Ylm',
    'Zlm',
]
